src/vmm/src/resources.rs

```diff
@@ -187,6 +189,23 @@ impl VmResources {
             return Err(VmConfigError::InvalidMemorySize);
         }
 
+        // The VM cannot have a memory size greater than the target size
+        // of the balloon device, if present.
+        if self.balloon.get().is_some() {
```

Comment:
What if the UX did not allow configuring the balloon before the machine_config ?

Reply:
machine-cfg has default values so it's optional, can't gate balloon on that. You could do it the other way around: can't config machine-cfg _after_ balloon, but I don't see it as an improvement to code in PR.

Reply:
I also believe we shouldn't introduce this kind of state inter-dependency between the balloon and machine config. I thought about it but, in a RESTful context, it _feels_ wrong to force users to send one **optional** machine config request if they want to also add a balloon.